python/HeaderService/hslib_salobj.py

```diff
@@ -475,7 +475,23 @@ def collect_from_HeaderService(self):
             self.metadata['RAEND'] = ra
             self.metadata['DECEND'] = dec
         else:
-            self.log.info("No 'ELEND' and 'AZEND'")
+            self.log.info("No 'ELEND' and 'AZEND' to compute RA/DEC END")
+
+        if set(('RA', 'DEC', 'ROTPA')).issubset(self.metadata):
+            # --- TODO ----
+            # In the near future, once radesys comes from telemetry,
+            # we should use self.metadata['RADESYS'] instead of the
+            # hardcorded FK5 value we pluged here.
+            # --------------
+            self.log.info("Computing WCS-TAN from RA/DEC/ROTPA")
+            wcs_TAN = self.HDR.CCDGEOM.wcs_TAN(self.metadata['RA'],
+                                               self.metadata['DEC'],
+                                               self.metadata['ROTPA'],
+                                               'FK5')
+            # Update the metadata
+            self.metadata.update(wcs_TAN)
+        else:
+            self.log.info("No 'RA', 'DEC' or 'ROTPA' to compute WCS")
```

Comment:
Please be specific about which cards are missing

Reply:
Fixed at requested.